backend/drm/drm.c

```diff
@@ -1616,3 +1628,73 @@ void destroy_drm_connector(struct wlr_drm_connector *conn) {
 	wl_list_remove(&conn->link);
 	free(conn);
 }
+
+int wlr_drm_backend_create_lease(struct wlr_backend *backend,
+		struct wlr_output *output, uint32_t *lessee_id) {
+	assert(backend && wlr_backend_is_drm(backend));
+	assert(output && wlr_output_is_drm(output));
+
+	struct wlr_drm_backend *drm = get_drm_backend_from_backend(backend);
+
+	struct wlr_drm_connector *connector = (struct wlr_drm_connector *)output;
+	assert(connector->state != WLR_DRM_CONN_LEASED);
+
+	if (connector->backend != drm) {
+		wlr_log(WLR_ERROR, "output doesn't belong to the backend");
+		return -EINVAL;
+	}
+
+	uint32_t objects[] = {
+		connector->id,
+		connector->crtc->id,
+		connector->crtc->primary->id,
+		(connector->crtc->cursor != NULL ? connector->crtc->cursor->id : 0),
+	};
+
+	int nobjects = sizeof(objects) / sizeof(objects[0]);
+
+	wlr_log(WLR_DEBUG, "Issuing DRM lease with the %d objects:", nobjects);
+	int lease_fd = drmModeCreateLease(drm->fd, objects, nobjects, 0, lessee_id);
+	if (lease_fd < 0) {
+		return lease_fd;
+	}
+
+	wlr_log(WLR_DEBUG, "Issued DRM lease %d", *lessee_id);
+	connector->lessee_id = *lessee_id;
+	connector->crtc->lessee_id = *lessee_id;
+	connector->state = WLR_DRM_CONN_LEASED;
+	wlr_output_destroy(&connector->output);
```

Comment:
Calling this mutates the connector state. I'd assume the new state to be `WLR_DRM_CONN_DISCONNECTED`.